### PR TITLE
Fix navigating back from Settings screens and refreshing SearchPage with policyIDs in the url

### PR DESCRIPTION
**Pull request by WojtekBoman**
### Details

This PR fixes 2 issues: 

1.  Refreshing the SearchPage with the selected workspace doesn't reset the workspace icon to its global state. This issue was caused by working directly on `topmostCentralPaneRoute.params` instead of on a copy of this object in `getMatchingBottomTabRouteForState.ts`

https://github.com/Expensify/App/assets/47774969/a20a3e04-8984-4950-abcc-8d2d9af262f8

2.  Navigating back from settings screen no longer pushes a new `Settings_Root` to the navigation state. This problem was caused by comparing paths in `getDistanceFromPathInRootNavigator`

https://github.com/Expensify/App/assets/47774969/d6a71313-fb4f-475b-9e0e-d8537dc6e4b8

### Fixed Issues

$ https://github.com/Expensify/App/issues/44478
PROPOSAL:

### Tests

- [ ] Verify that no errors appear in the JS console

1. First issue:

- Select any workspace on SearchPage
- Refresh the page
- Verify if the workspace switcher icon matches the selected workspace.

2. Second issue:

- Launch the app
- Tap Settings
- Tap About
- Tap Back arrow
- Verify if navigating back to the Settings works properly and the animation is correct (The about page animated from left to right) 

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/47774969/735fe337-a7bb-46be-8473-d95d9fc58b3a

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/47774969/2840aa98-6bd4-4d95-b429-8fa82306fea5

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/47774969/ccec2b0d-ddff-4606-8449-902bbf1dc159

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/47774969/857cf155-d730-4854-add8-f831425a9ddc

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/47774969/9367b308-6d3c-444f-ab29-766f5df7560c

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/47774969/1940572a-68b6-4c02-9a16-23e9a1782007

</details>

### CODE REVIEW

File: src/libs/Navigation/Navigation.ts [hunk 1]
```diff
@@ -129,7 +129,9 @@ function getDistanceFromPathInRootNavigator(path?: string): number {
             break;
         }
 
-        const pathFromState = getPathFromState(currentState, linkingConfig.config);
+        // When paths are compared, they the policyID param should be excluded from the state. If it's present in currentState, pathFromState will include it and in result it will be different than the path value.
```
Comment (Kicu) [inline src/libs/Navigation/Navigation.ts:132]: `they the ` please fix comment.
I would also shorten the comment a bit, I think its hard to parse because it has a lot of details. Just make it a little bit simpler

File: src/libs/NavigationUtils.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import _ from 'lodash';
```
Comment (Kicu) [inline src/libs/NavigationUtils.ts:1]:
```suggestion
import cloneDeep from 'lodash/cloneDeep';
```

Try yo never import directly from lodash. In the long run it should be possible to remove the full lodash from project, and import only some of its specific packages

**Review by Kicu — APPROVED** [review]
Looks nice, good job 👍 left you 2 small comments

**Comment by WojtekBoman** [comment]
> @Kicu @WojtekBoman Do we need C+ here? I see the original issue is not open yet. Thanks

The original issue is open here: https://github.com/Expensify/App/issues/44478 and we need C+ to review this :)

**Review by dukenv0307 — APPROVED** [review]

**Review by mountiny — APPROVED** [review]
Thanks
